sender.go

```diff
@@ -26,6 +31,12 @@ type Sender struct {
 // additional messages can be sent while the current goroutine is waiting
 // for the confirmation.
 func (s *Sender) Send(ctx context.Context, msg *Message) error {
+	err := s.link.Check()
```

Comment:
Nit: condense into the `if` block as `err` is immediately returned.
```go
if err := s.link.Check(); err != nil {
    return err
}
```

Reply:
Done!